pkg/gui/controllers/status_controller.go

```diff
@@ -68,26 +68,15 @@ func (self *StatusController) GetKeybindings(opts types.KeybindingsOpts) []*type
 }
 
 func (self *StatusController) GetOnRenderToMain() func() error {
-	return func() error {
-		dashboardString := strings.Join(
-			[]string{
-				lazygitTitle(),
-				"Copyright 2022 Jesse Duffield",
-				fmt.Sprintf("Keybindings: %s", constants.Links.Docs.Keybindings),
-				fmt.Sprintf("Config Options: %s", constants.Links.Docs.Config),
-				fmt.Sprintf("Tutorial: %s", constants.Links.Docs.Tutorial),
-				fmt.Sprintf("Raise an Issue: %s", constants.Links.Issues),
-				fmt.Sprintf("Release Notes: %s", constants.Links.Releases),
-				style.FgMagenta.Sprintf("Become a sponsor: %s", constants.Links.Donate), // caffeine ain't free
-			}, "\n\n")
-
-		return self.c.RenderToMainViews(types.RefreshMainOpts{
-			Pair: self.c.MainViewPairs().Normal,
-			Main: &types.ViewUpdateOpts{
-				Title: self.c.Tr.StatusTitle,
-				Task:  types.NewRenderStringTask(dashboardString),
-			},
-		})
+	config := self.c.UserConfig.Gui
+
+	switch config.DefaultMainView {
+	case "dashboard":
+		return self.showDashboard
+	case "allBranchesLog":
+		return self.showAllBranchLogs
+	default:
+		return self.showDashboard
```

Comment:
Something like this?
```go
default:
	return func() error {
		return self.c.Error(fmt.Error("Unexpected value for StatusPanelView config"));
	}
```

Reply:
True, but I'm not sure what else to do. In other cases we panic at the place where it's used (e.g. for [`git.log.showGraph`](https://github.com/jesseduffield/lazygit/blob/b01bad7fad62ec54616d9cb9f0958da1d65c7aac/pkg/gui/context/local_commits_context.go#L159), when it was still a config), and that's a terrible UX, we shouldn't do that.

We could validate the config value right after loading, but then there's the risk that the validation code runs out of sync with the code that uses it, since we're working with string literals everywhere.

Reply:
Oops, our messages crossed. Yes, that would also work, but I think I'd prefer to do the validation right after loading (i.e. at the end of `loadUserConfig`.

Reply:
>We could validate the config value right after loading, but then there's the risk that the validation code runs out of sync with the code that uses it, since we're working with string literals everywhere.

:grimacing: you're right.  Maybe we do just fallback to default then. We have the schema to at least help the user out here anyway.

Reply:
Added `UserConfig` validation. In particular validation for `Git.StatusPanelView`.